File: benchmarks/src/python/word_count.py
import time
M=500000;seed=12345
def ri(m):
 global seed;seed=(seed*16807)%2147483647;return seed%m
d=['w'+str(i) for i in range(2000)]
c={w:0 for w in d}
start=time.time()*1000
for _ in range(M):c[d[ri(2000)]]+=1
mc=0;mw=''
for w in d:
 v=c[w]
 if mc==0:mc=v;mw=w
 elif v>mc:mc=v;mw=w
end=time.time()*1000
print(f"Result: {mc}_{mw}")
print(f"Time: {int(end-start)}ms")

File: benchmarks/src/ruby/word_count.rb
M=500000;seed=12345;def ri(m);$seed=($seed*16807)%2147483647;$seed%m;end;$seed=12345
dict=(0...2000).map{|i|"w#{i}"}
counts={};dict.each{|w|counts[w]=0}
start=Time.now.to_f*1000
M.times{counts[dict[ri(2000)]]+=1}
mc=0;mw="";dict.each{|w|v=counts[w];mc=mw==""||v>mc ?(mc=v;mw=w;mc):mc}
finish=Time.now.to_f*1000
puts"Result: #{mc}_#{mw}";puts"Time: #{(finish-start).round}ms"
